### PR TITLE
add: Add deployment and daemonset user guide 

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: Harrison harrisonliu_5@163.com
/assign @leoendless @FeynmanZhou

### CODE REVIEW

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 1]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_form_2.png)
+
+#### Step 3.1: Container Setting
+
+Click the **Add Container Image** area.
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_form_2_container_btn.png)
+
+You can input or select the image which in the public docker hub or private image repository you want to use.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:39]: ->
You can input an image name to use the image from public Docker Hub or select an image from a private repository you want to use.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 2]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_form_2_container_1.png)
+
+- **Image Name**
+
+  You can click the icon or input the image name to search it. kubesphere provide the Dockerhub image and your private image repository. If you want to use your private image repository, you should create a docker hub secret first in **Configurations** **Secrets**
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:45]: ->
You can click the cube icon or input the image name to search it. KubeSphere provides Docker Hub images and your private image repository. If you want to use your private image repository, you should create a Docker Hub secret first in **Secrets** under **Configurations**.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 3]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+- **Image Tag**
+
+  You can input tag like `imagename:tag`, If you do not specify, it will default to the latest version.
+
+- **Container Type**
+
+  Choose Init Container，It means that to create the Init Container。 For more information about Init Container, please visit [Init Container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/?spm=a2c4g.11186623.2.19.16704b3e9qHXPb)
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:53]: ->
Choose Init Container, which means the init container will be created for the workload. For more information about init containers, please visit [Init Container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/?spm=a2c4g.11186623.2.19.16704b3e9qHXPb).

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 4]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_form_2_container_2.png)
+
+- **Resource Request**
+
+  The resource quota reserved by the container includes both CPU and memory resources. That is the container monopolizes the resource, preventing other services or processes from competing for resources due to insufficient resources, causing the application to become unavailable.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:59]: That is the container monopolizes the resource
->
It means the container monopolizes the resource

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 5]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+  - name：Set the name of the environment variable.
+  - value：Set the value of the variable reference.
+  - type：Set the type of environment variables, support customization, configuration items, keys, and variable/variable references.
+
+  For more information about the command, please visit [Pod variable](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/?spm=a2c4g.11186623.2.20.16704b3e9qHXPb).
+
+- **Container Security Context**
+
+  A security context defines privilege and access control settings for a Pod or Container. [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:148]: [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/)
->
For more information about the security context, see [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 6]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+{{</ notice>}}
+
+- **Pod Security Context**
+
+  A security context defines privilege and access control settings for a Pod or Container. For more information about Pod Security Policies, please visit [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
+
+- **Deployment Mode**
+
+  Select different deployment modes to change the affinity and anti-affinity of the inter-pod. In k8s, the affinity is used `podAffinity` and the anti-affinity is used `podAntiAffinity`. In the Kubesphere both affinity and anti-affinity are set `preferredDuringSchedulingIgnoredDuringExecution`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:189]: ->
You can select different deployment modes to switch between inter-pod affinity and inter-pod anti-affinity. In Kubernetes, inter-pod affinity is specified as field `podAffinity` of field `affinity` while inter-pod anti-affinity is specified as field `podAntiAffinity` of field `affinity`. In KubeSphere, both `podAffinity` and `podAntiAffinity` are set to `preferredDuringSchedulingIgnoredDuringExecution`. You can enable **Edit Mode** in the top right corner to see field details.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 7]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+  - Pod Decentralized Deployment is mean anti-affinity
+  - Pod Aggregation Deployment is mean affinity
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:192]: ->
**Pod Decentralized Deployment** represents anti-affinity.
**Pod Aggregation Deployment** represents affinity.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 8]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+{{</ notice>}}
+
+### Step 5: Input Advanced Data
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_form_4.png)
+
+- **Set Node Scheduling Policy**
+
+  You can allow Pod replicas to run on specified nodes. It is used `nodeSelector`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:242]: It is used `nodeSelector`.
->
It is specified in the field `nodeSelector`.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 9]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+- **Add Metadata**
+
+  Additional metadata settings for resources such as **Labels** and **Annotations**.
+
+## Check Deployment Detail
+
+You can check the deployment detail via click deployment's name on the list.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:250]: ->
You can check the Deployment detail by clicking the Deployment's name in the list.

File: content/en/docs/project-user-guide/application-workloads/deployments.md [hunk 10]
```diff
@@ -7,4 +7,291 @@ description: 'Kubernetes Deployments'
...
+
+### Deployment Operations
+
+![deployments](/images/docs/project-user-guide/workloads/deployments_list.png)
+
+You can click **More** to display what other operations about this Deployment you can do.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/deployments.md:256]: ->
Click the three dots on the right and select the operation from the menu to modify a Deployment.

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments and make changes. Thanks.

**Comment by harrisonliu5** [comment]
Has been modified

**Comment by Sherlock113** [comment]
/lgtm
@FeynmanZhou Please merge this pr. I will update wording and refactor these two guides with the statefulset guide as some content is entirely identical.

**Pull request merged**